rush-plugins/rush-amazon-s3-build-cache-plugin/src/RushAmazonS3BuildCachePlugin.ts

```diff
@@ -17,14 +17,14 @@ const PLUGIN_NAME: string = 'AmazonS3BuildCachePlugin';
  */
 export interface IAmazonS3ConfigurationJson {
   /**
-   * The Amazon S3 region of the bucket to use for build cache (e.g. "us-east-1").
+   * The Amazon S3 endpoint of the bucket to use for build cache (e.g. "s3.us-east-2.amazonaws.com").
    */
-  s3Region: string;
+  s3Endpoint: string;
 
   /**
-   * The name of the bucket in Amazon S3 to use for build cache.
+   * The Amazon S3 region of the bucket to use for build cache (e.g. "us-east-1").
    */
-  s3Bucket: string;
```

Comment:
I'm not nearly as familiar with S3 as I am with Azure Storage, but isn't the bucket still required? Isn't the bucket equivalent to the Azure container?

Reply:
The bucket name is resolved using the hostname, so it is not needed.

(the bucket name can also be resolved using the pathname, but this is deprecated by AWS, https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/)